tests/nfs/nfs_verify_file_operations.py

```diff
@@ -0,0 +1,120 @@
+from threading import Thread
+from time import sleep
+
+from nfs_operations import cleanup_cluster, setup_nfs_cluster
+
+from cli.exceptions import ConfigError
+from cli.utilities.utils import (
+    change_ownership,
+    change_permission,
+    create_files,
+    perform_lookups,
+)
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """Verify create file, create soflink and lookups from nfs clients
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    nfs_nodes = ceph_cluster.get_nodes("nfs")
+    clients = ceph_cluster.get_nodes("client")
+
+    port = config.get("port", "2049")
+    version = config.get("nfs_version", "4.0")
+    no_clients = int(config.get("clients", "2"))
+    file_count = int(config.get("file_count", "10"))
+    operations = config.get("operations")
+    Thread_operations = []
+    permissions = "+rwx"
+    user = "cephuser"
+
+    # If the setup doesn't have required number of clients, exit.
+    if no_clients > len(clients):
+        raise ConfigError("The test requires more clients than available")
+
+    clients = clients[:no_clients]  # Select only the required number of clients
+    nfs_node = nfs_nodes[0]
+    fs_name = "cephfs"
+    nfs_name = "cephfs-nfs"
+    nfs_export = "/export"
+    nfs_mount = "/mnt/nfs"
+    fs = "cephfs"
+    nfs_server_name = nfs_node.hostname
+
+    try:
+        # Setup nfs cluster
+        setup_nfs_cluster(
+            clients,
+            nfs_server_name,
+            port,
+            version,
+            nfs_name,
+            nfs_mount,
+            fs_name,
+            nfs_export,
+            fs,
+        )
+
+        # Create oprtaions on each client
+        for client, operation in operations.items():
+            if operation == "create_files":
+                Thread_operations.append(
+                    Thread(
+                        target=create_files,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount, file_count),
+                    )
+                )
+            elif operation == "change_ownership":
+                Thread_operations.append(
+                    Thread(
+                        target=change_ownership,
+                        args=(
+                            clients[int(client[-2:]) - 1],
+                            nfs_mount,
+                            file_count,
+                            user,
+                        ),
+                    )
+                )
+            elif operation == "perform_lookups":
+                Thread_operations.append(
+                    Thread(
+                        target=perform_lookups,
+                        args=(clients[int(client[-2:]) - 1], nfs_mount, file_count),
+                    )
+                )
+            elif operation == "change_permission":
```

Comment:
Currently as per logs, the change_permission did not ran and gave below error in logs -

-------
2023-07-25 03:32:47,493 (cephci.nfs_verify_file_operations) [ERROR] - cephci.ceph.ceph.py:1594 - Error 1 during cmd, timeout 600
2023-07-25 03:32:47,494 (cephci.nfs_verify_file_operations) [ERROR] - cephci.ceph.ceph.py:1595 - chmod: cannot access '/mnt/nfs/file1': No such file or directory
---------
A slight delay or sleep is required between create file and change permission function as Client 2 is changing permission of file 1 when Client 1 is still under process of creating new file via dd

Reply:
Thanks @msainiRedhat 
Slight delay already provided

https://github.com/red-hat-storage/cephci/pull/2758/files#diff-c0cd182f3c1ff023ea86528e17222be3a1f092e8581489a86a82ebdbc85d0241R107

Please find correct logs
Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-XARFTN/